File: algorithms/sorting/insertion sort/insertionSort.js
const fs = require('fs');
const values = fs
  .readFileSync('values.txt', 'utf8')
  .split(',')
  .map((x) => Number(x));

function insertionSort(values) {
  for (let i = 0; i < values.length; i++) {
    let current = values[i];
    let j = i - 1;
    while (j >= 0 && values[j] > current) {
      values[j + 1] = values[j];
      j--;
    }
    values[j + 1] = current;
  }
  return values;
}

const sorted = insertionSort(values);
console.log(sorted);

File: algorithms/sorting/insertion sort/insertion_sort.py
data = open("values.txt", "r")
values = [int(v) for v in data.read().split(",")]


def insertionSort(values):
    for i in range(len(values)):
        current = values[i]
        j = i - 1
        while j >= 0 and values[j] > current:
            values[j + 1] = values[j]
            j -= 1
        values[j+1] = current
    return values


result = insertionSort(values)
print(result)
